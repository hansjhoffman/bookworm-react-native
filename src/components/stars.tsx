import React from "react";
import { Image, StyleSheet, View } from "react-native";

import { Icons } from "../../assets";

type Props = {
  rating: number;
};

export function Stars({ rating }: Props) {
  let stars = [];

  for (let idx = 0; idx < 5; idx++) {
    stars.push(
      <Image
        key={idx}
        source={Icons.star}
        style={{
          width: 18,
          height: 18,
          tintColor: idx < rating ? "#f4ca67" : "#dbdbdb",
          marginLeft: idx > 0 ? 1.5 : 0,
        }}
      />,
    );
  }

  return <View style={styles.container}>{stars}</View>;
}

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
  },
});
