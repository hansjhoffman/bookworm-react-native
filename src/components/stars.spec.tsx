import React from "react";
import { render } from "@testing-library/react-native";

import { Stars } from "./stars";

describe("[component] Stars", function() {
  it("should match snapshot", function() {
    const { baseElement } = render(<Stars rating={5} />);

    expect(baseElement).toMatchSnapshot();
  });
});
