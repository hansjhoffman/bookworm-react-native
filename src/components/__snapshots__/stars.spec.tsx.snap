// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[component] Stars should match snapshot 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/icons/star.png",
          }
        }
        style={
          Object {
            "height": 18,
            "marginLeft": 0,
            "tintColor": "#f4ca67",
            "width": 18,
          }
        }
      />
      <Image
        source={
          Object {
            "testUri": "../../../assets/icons/star.png",
          }
        }
        style={
          Object {
            "height": 18,
            "marginLeft": 1.5,
            "tintColor": "#f4ca67",
            "width": 18,
          }
        }
      />
      <Image
        source={
          Object {
            "testUri": "../../../assets/icons/star.png",
          }
        }
        style={
          Object {
            "height": 18,
            "marginLeft": 1.5,
            "tintColor": "#f4ca67",
            "width": 18,
          }
        }
      />
      <Image
        source={
          Object {
            "testUri": "../../../assets/icons/star.png",
          }
        }
        style={
          Object {
            "height": 18,
            "marginLeft": 1.5,
            "tintColor": "#f4ca67",
            "width": 18,
          }
        }
      />
      <Image
        source={
          Object {
            "testUri": "../../../assets/icons/star.png",
          }
        }
        style={
          Object {
            "height": 18,
            "marginLeft": 1.5,
            "tintColor": "#f4ca67",
            "width": 18,
          }
        }
      />
    </View>
  </View>
</View>
`;
